packages/wrangler/src/git-client.ts

```diff
@@ -40,3 +53,72 @@ export async function initializeGit(cwd: string) {
 		});
 	}
 }
+
+/**
+ * Clone a repository into a given directory, optionally applying a given filter for a sparse checkout.
+ * Note: this does NOT initialize a git repository, that must be done separately.
+ *
+ * @param remote the remote repository to clone
+ * @param targetDirectory the directory to clone into
+ * @param subdirectory optional, clone a subdirectory instead of the whole repo
+ */
+export async function cloneIntoDirectory(
+	remote: string,
+	targetDirectory: string,
+	subdirectory?: string
+) {
+	const args = ["clone", "--depth", "1"];
+
+	const gitVersion = await isGitInstalled();
+	if (!gitVersion) {
+		throw new Error("Failed to find git installation");
+	}
+
+	// sparse checkouts were added in git 2.26.0, and allow for...sparse...checkouts...
+	// basically it means you can checkout only certain files/folders/etc. instead of
+	// the whole repo. useful for monorepos that may, for example,
+	// contain multiple wrangler templates
+	const useSparseCheckout = semiver(gitVersion, "2.26.0") > -1;
```

Comment:
Perhaps even cleaner:

```suggestion
	const useSparseCheckout = subdirectory && semiver(gitVersion, "2.26.0") > -1;
```

Then we don't need to remember to keep checking the subdirectory whenever we use `useSparseDirectory`.